### PR TITLE
feat: add user admin middleware and tests

### PR DESCRIPTION
**Pull request by Sofiag8**
## Describe your changes
User admin middleware added with tests. This middleware will verify that events are created, updated and deleted only by admin users

note: I also  refactor auth-user-token tests to encapsulate the way proxyrequire and stubs are use. 

## Issue ticket number and link
https://trello.com/c/90RclDFa/145-crear-custom-middleware-para-verificar-que-el-usuario-es-admin

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

### CODE REVIEW

**Review by FelipeLosada1996 — APPROVED** [review]
:+1:

**Review by jose-sal-y-rosas — CHANGES_REQUESTED** [review]
Please review comment

File: middlewares/auth-user-admin.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+async function authenticateAdminUser(req, res, next) {
+  try {
+    const userService = await serviceContainer('users')
+    const userData = await userService.findById(req.user.id)
+
+    if (!userData.data.isAdmin) {
+      return res.status(401).json({
+        status: '401',
+        message: 'Unauthorized, to perform this action admin role is required',
```
[the remark this answers is not in the document]
Comment (Sofiag8) [reply middlewares/auth-user-admin.js:12]: TODO added. Trello task: https://trello.com/c/qios6ThS/150-crear-archivo-de-constantes-para-manejo-de-errores

**Review by jose-sal-y-rosas — APPROVED** [review]
👍🏼

**Pull request merged**